plugins/include-markdown/index.js

```diff
@@ -32,8 +36,16 @@ module.exports = function includeMarkdownPlugin({ resolveFrom } = {}) {
       // if any other file type, they are embedded into a code block
       if (includePath.match(/\.md(?:x)?$/)) {
         // return the file contents in place of the @include
-        // this takes a couple steps because we allow recursive includes
-        const processor = remark().use(includeMarkdownPlugin, { resolveFrom })
+        // (takes a couple steps because we're processing includes with remark)
+        const processor = remark()
+        // if the include is MDX, and the plugin consumer has confirmed their
+        // ability to stringify MDX nodes (eg "jsx"), then use remarkMdx to support
+        // custom components (which would otherwise appear as likely invalid HTML nodes)
+        const isMdx = includePath.match(/\.mdx$/)
+        if (isMdx && resolveMdx) processor.use(remarkMdx)
```

Comment:
The `resolveMdx` option is opt-in, because otherwise, use of this plugin could potentially break use cases where `remark-stringify` is used. This is because `remark-mdx` will produce `jsx`-type nodes for custom components, which `remark-stringify` will choke on.